aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsProxyHttpServletRequest.java

```diff
@@ -701,7 +702,7 @@ public AsyncContext startAsync(ServletRequest servletRequest, ServletResponse se
         try {
             List<FileItem> items = upload.parseRequest(this);
             for (FileItem item : items) {
-                String fileName = SecurityUtils.getValidFilePath(item.getName(), true);
+                String fileName = FilenameUtils.getName(item.getName());
```

Comment:
Reading the FindBugs docs, it says that if you are using Java7u40 or JAva 8, you don't have to worry about null byes in filenames any more.
Ref: https://find-sec-bugs.github.io/bugs.htm

Reply:
I might also argue for leraving the string provided by the client as is, since trying to use this path in a lambda (or any other situation) doesn't make much sense to me.

Reply:
I heard about the fix on the null bytes, just concerned about regressions - although it may be overkill. What's the behavior if the file name comes through as null? Is it even possible? I need to look into it.

Reply:
So I did, turns out that all of these checks are already performed when you call `item.getName()`:

```java
public static String checkFileName(String fileName) {
        if (fileName != null  &&  fileName.indexOf('\u0000') != -1) {
            // pFileName.replace("\u0000", "\\0")
            final StringBuilder sb = new StringBuilder();
            for (int i = 0;  i < fileName.length();  i++) {
                char c = fileName.charAt(i);
                switch (c) {
                    case 0:
                        sb.append("\\0");
                        break;
                    default:
                        sb.append(c);
                        break;
                }
            }
            throw new InvalidFileNameException(fileName,
                    "Invalid file name: " + sb);
        }
        return fileName;
    }
```

If the client sent a file name as null, we probably shouldn't do anything and expose the issue to the developer rather than hide this ourselves.